### ISSUE TITLE
CSS Variables With React Spring

### ISSUE THREAD

**Issue by BryanAM** [pmndrs/react-spring#1651]
## 🚀 Feature Proposal

react-spring should accept css variables as props and animate the same as if defined explicitly in the spring.

## Motivation

Theming requires styles and colors depending on theme type and using css vars is a common and easy way to map different colors & themes to one css variable. 

```
:root { 
 .default-theme {
   --box-shadow: 0px 0px 31px 1px rgba(98, 0, 238,1);
  }

  .dark-theme {
     --box-shadow: 0px 0px 10px 3px rgba(113,113,113,0.4);
  }

  .special-theme {
     --box-shadow: 0px 2px 10px 3px rgba(98,98, 238,1);
  }
}
```

## Example

I have a component that inherits box-shadow variables depending on the theme. The box-shadow should animate onHover.
If the box shadow is defined as a string, the animation works as expected. However, if I use my css variable - which works without animations - animations are broken. 

```
import React from 'react';
import PropTypes from 'prop-types';
import { useSpring, animated } from 'react-spring';
import './search.scss';

const Search = ({ change }) => {
  const [animation, api] = useSpring(() => ({
    from: { boxShadow: 'var(--box-shadow)' } }));

  return (
    <>
      <animated.input
        placeholder="🔎 Search..."
        className="search"
        onChange={change}
        style={animation}
        onMouseOver={() => api.start({ boxShadow: 'var(--box-shadow-hover)' })}
        onMouseLeave={() => api.start({  boxShadow: 'var(--box-shadow)' })}
        onFocus={() => console.log('focus')}
      />
    </>
  );
};

Search.propTypes = {
  change: PropTypes.func.isRequired,
};

export default Search;
```

### Problem 
Without being able to use css variables, the animating colors cannot be dynamically rendered depending on the theme.

**Comment by joshuaellis** [comment]
Thanks @BryanAM this looks like it would be a great addition! I guess we'd need to add some sort of layer to resolve the CSS var and get it's value before we can interpolate it and make sure we use the correct `AnimatedX` internally for it.

**Comment by joshuaellis** [comment]
This has been released on `9.4.0-beta.6`
